model_compression_toolkit/core/common/substitutions/shift_negative_activation.py

```diff
@@ -476,16 +421,15 @@ def shift_negative_function(graph: Graph,
 
     if len(previous_nodes) == 1:
         fused_candidates = [previous_nodes[0], non_linear_node, add_node]
+        graph.fusing_info.fusing_patterns.append([n.type for n in fused_candidates])
+        fused_op_id = graph.fusing_info.generate_fused_op_id(fused_candidates)
 
-        if graph.fusing_info.is_nodes_eligible_to_be_fused(fused_candidates):
-            fused_op_id = graph.fusing_info.generate_fused_op_id(fused_candidates)
-
-            # If the non-linear is already part of a fused op - remove the existing one before adding the new one
-            if graph.fusing_info.is_node_in_fused_op(non_linear_node):
-                existing_fused_op = graph.fusing_info.get_fused_node_name(non_linear_node.name)
-                graph.fusing_info.remove_fused_operation(existing_fused_op)
+        # If the non-linear is already part of a fused op - remove the existing one before adding the new one
+        if graph.fusing_info.is_node_in_fused_op(non_linear_node):
```

Comment:
what's the scenario for this?

Reply:
If for example there is a fusion for 'conv->swish', but adding the 'add' should replace it to a new fusion of 'conv->swish->add'